### PR TITLE
Enable user to edit inventory list items from the front end

### PR DESCRIPTION
**Pull request by danascheider**
## Context

[**Enable user to edit inventory list items from the front end**](https://trello.com/c/bF5ivF9S/150-enable-user-to-edit-inventory-list-items-from-the-front-end)

Users still are not able to edit inventory list items from the UI. Editing includes incrementing and decrementing the quantity as well as editing other attributes using the form.

## Changes

* Create `InventoryListItemEditForm` component to be displayed in modal
* Add new edit icon to inventory list item to trigger the form
* Add new increment/decrement icons to inventory list item to increment or decrement quantity
* Add `updateInventoryListItem` and `destroyInventoryListItem` functions to the `simApi` module to handle those API calls
* Add `performInventoryListItemUpdate` and `performInventoryListItemDestroy` functions to the `InventoryListsContext` to update and destroy items and handle the responses
* Add missing tests for shopping list item editing (primarily for cases where the user decrements a quantity to zero and chooses to destroy the item when prompted)
* Fix bug where unit weight was being sent to the back end as an empty string sometimes
* Add tests for new functionality
* Update context docs

## Considerations

This is again a lot of duplicated code from shopping list items that will possibly be refactored before this epic is considered complete.

### Edit Form, 1201px

Screen Shot 2021-08-21 at 7 59 19 am

### Edit Form, 1025px

Screen Shot 2021-08-21 at 7 59 34 am

### Edit Form, 769px

Screen Shot 2021-08-21 at 7 59 48 am

### Edit Form, 601px

Screen Shot 2021-08-21 at 7 59 59 am

### Edit Form, 425px

Screen Shot 2021-08-21 at 8 00 26 am

### Inventory Lists Page, 1201px

issue-150_main-page_1201px

### Inventory Lists Page, 1025px

issue-150_main-page_1025px

### Inventory Lists Page, 769px

issue-150_main-page_769px

### Inventory Lists Page, 601px

issue-150_main-page_601px

### Inventory Lists Page, 425px

issue-150_main-page_425px

### CODE REVIEW

File: src/components/inventoryList/inventoryList.js [hunk 1]
```diff
@@ -185,7 +185,7 @@ const InventoryList = ({ canEdit = true, listId, title }) => {
                   itemId={id}
                   listTitle={title}
                   description={description}
-                  quantity={quantity}
+                  quantity={parseInt(quantity)}
```
Comment (danascheider) [inline src/components/inventoryList/inventoryList.js:188]: This kept giving errors that the quantity being passed in was a string.

File: src/components/shoppingListItem/shoppingListItem.js [hunk 2]
```diff
@@ -119,6 +122,8 @@ const ShoppingListItem = ({
     const oldQuantity = currentQuantity
     const newQuantity = currentQuantity - 1
 
+    if (mountedRef.current) setCurrentQuantity(newQuantity)
```
Comment (danascheider) [inline src/components/shoppingListItem/shoppingListItem.js:125]: This was accidentally left out when I originally wrote this code.

File: src/components/shoppingListItem/shoppingListItem.js [hunk 3]
```diff
@@ -204,7 +210,7 @@ const ShoppingListItem = ({
       <div className={styles.toggle} onClick={toggleDetails}>
         <span className={classNames(styles.header, { [styles.headerEditable]: canEdit })}>
           {canEdit &&
-            <span className={styles.editIcons} ref={iconsRef}>
```
Comment (danascheider) [inline src/components/shoppingListItem/shoppingListItem.js:207]: `iconsRef` was never actually used.

File: src/components/shoppingListsPageContent/shoppingListsPageContent.stories.js [hunk 4]
```diff
@@ -235,7 +235,7 @@ HappyPath.parameters = {
         // found in any list's array of list items. The list item doesn't exist
         // or doesn't belong to the authenticated user.
         return res(
-          ctx.status(204)
```
Comment (danascheider) [inline src/components/shoppingListsPageContent/shoppingListsPageContent.stories.js:238]: Oops

File: src/components/inventoryListItem/inventoryListItem.js [hunk 5]
```diff
@@ -18,28 +23,63 @@ const formatWeight = weight => {
 }
 
 const InventoryListItem = ({
-  canEdit = true,
   itemId,
+  listTitle,
```
Comment (danascheider) [inline src/components/inventoryListItem/inventoryListItem.js:27]: Reorganised these to make it more like the `ShoppingListItem` component so it's easier to compare the two components when we want to refactor/extract common code.
